Parse/src/main/java/com/parse/ParseQuery.java

```diff
@@ -1699,6 +1701,31 @@ public void getInBackground(final String objectId, final GetCallback<T> callback
    * {@code ParseQuery}.
    * <p/>
    * This only works on keys whose values are {@link ParseObject}s or lists of {@link ParseObject}s.
+   * Add a constraint to the query that requires a particular key's value to contain each one of
+   * the provided list of values entirely or just starting with given strings.
+   *
+   * @param key
+   *          The key to check. This key's value must be an array.
+   * @param values
+   *          The values that will match.
+   * @return this, so you can chain this call.
+   */
+  public ParseQuery<T> whereContainsAllStartsWith(String key, Collection<String> values) {
```

Comment:
Could be a good name, but I think it's good to have the `All` word to know that all strings in collection must match. But I like your option too.

Let's see what other prefer.